### PR TITLE
feat: add `iter/cuany-by`

### LINKED ISSUE #2335: [RFC]: add `@stdlib/iter/cuany-by`
### Description

This RFC proposes adding the package `@stdlib/iter/cuany-by`, which cumulatively tests whether at least one iterated value passes a test implemented by a predicate function. The returned iterator should be a transform iterator, continuing to iterate while source iterator values are available.

```javascript
var array2iterator = require( '@stdlib/array/to-iterator' );

function isPositive( value ) {
	return ( value > 0 );
}

var arr = array2iterator( [ 0, 0, 0, 1, 0 ] );

var it = iterCuAnyBy( arr, isPositive );

var v = it.next().value;
// returns false

v = it.next().value;
// returns false

v = it.next().value;
// returns false

v = it.next().value;
// returns true

v = it.next().value;
// returns true

var bool = it.next().done;
// returns true
```

The `predicate` function should be provided two arguments:

- **value**: the iterated value.
- **index**: iteration index (zero-based).

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/iter/any-by`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by pra2107tham**
Resolves #2335  .

## Description

This pull request adds the `iterCuAnyBy` function to the `@stdlib/iter` package.  This function creates an iterator that cumulatively tests whether at least one iterated value passes a user-provided predicate function.

**Key features and changes:**

-   Implements the `iterCuAnyBy` function, returning an iterator of boolean values.
-   The predicate function receives the value and its index as arguments.
-   The returned iterator yields `false` until a value passes the predicate, then yields `true` for all subsequent values.
-   Includes comprehensive unit tests with various test cases.
-   Adds documentation for the function in the README and examples.
-   Updates benchmark tests to include the predicate function.

## Related Issues

This pull request resolves #2335.

## Questions

None at this time.

## Checklist

- [x] Read, understood, and followed the [contributing guidelines][contributing].

### CODE REVIEW

File: lib/node_modules/@stdlib/iter/cuany-by/docs/repl.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        iterated value (if one exists) and a boolean flag indicating whether the
+        iterator is finished.
+
+    iterator.return( [value] ): Function
+        Finishes an iterator and returns a provided value.
+
+    Examples
+    --------
+    > var arr = {{alias:@stdlib/array/to-iterator}}( [ 0, 0, 0, 1, 0 ] );
+    > var it = {{alias}}( arr );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuany-by/docs/repl.txt]: You need to update the example to use a predicate function. It is also missing from the API documentation above.

File: lib/node_modules/@stdlib/iter/cuany-by/examples/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+var opts = {
+	'iter': 100
+};
+var riter = randu( opts );
+
+// Create an iterator which applies a threshold to generated numbers:
+var miter = iterMap( riter, threshold );
+
+// Create an iterator which cumulatively tests whether at least one iterated value is truthy:
+var it = iterCuAnyBy( miter );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuany-by/examples/index.js]: Your example is missing a predicate function. You don't need `miter` here. You could pass `threshold` directly to `iterCuAnyBy`.

File: lib/node_modules/@stdlib/iter/cuany-by/docs/repl.txt (hunk 1 above)
Comment (pra2107tham) [reply lib/node_modules/@stdlib/iter/cuany-by/docs/repl.txt]: i checked it and updated, how to solve the lint errrors I am getting

Comment (pra2107tham) [reply lib/node_modules/@stdlib/iter/cuany-by/docs/repl.txt]: I am not able to figure out what to do next? can  you please tell me whate xacty and how to solve the issue

**Pull request closed without merging**